packages/core/test/acceptance/styling_spec.ts

```diff
@@ -2173,6 +2173,101 @@ describe('styling', () => {
        fixture.detectChanges();
        expect(div.classList.contains('disabled')).toBe(false);
      });
+
+  it('should throw an error if a prop-based style/class binding value is changed during checkNoChanges',
+     () => {
+       let updateStyleBinding = false;
+       let updateStyleCount = 0;
+       let updateClassBinding = false;
+       let updateClassCount = 0;
+
+       @Component({
+         template: `
+        <div [style.color]="color" [class.foo]="fooClass"></div>
+      `
+       })
+       class Cmp {
+         private _color = 'red';
+         private _fooClass = true;
+
+         get color() {
+           if (updateStyleBinding && updateStyleCount++) {
+             this._color = 'blue';
+           }
+           return this._color;
+         }
+
+         get fooClass() {
+           if (updateClassBinding && updateClassCount++) {
+             this._fooClass = false;
+           }
+           return this._fooClass;
+         }
+       }
+
+       TestBed.configureTestingModule({declarations: [Cmp]});
+       const fixture = TestBed.createComponent(Cmp);
+       fixture.detectChanges();
+
+       updateStyleBinding = true;
+       updateStyleCount = 0;
+
+       expect(() => { fixture.detectChanges(); }).toThrow();
```

Comment:
Let's check the specific error is a change detection error. Otherwise this test would pass if any arbitrary error is thrown.

```suggestion
       expect(() => { fixture.detectChanges(); }).toThrowError(/ExpressionChangedAfterItHasBeenCheckedError/);
```